include/deal.II/base/utilities.h

```diff
@@ -1548,6 +1578,15 @@ namespace Utilities
          T (&unpacked_object)[N],
          const bool allow_compression)
   {
+    // empty arrays are not allowed and so unpacking requires to
+    // have a non-zero amount of input data
+    Assert(
+      cbegin != cend,
+      ExcMessage(
+        "Utilities::unpack was asked to unpack "
+        "an array, but no input data to unpack was provided. Empty arrays are "
+        "not allowed."));
```

Comment:
This is ok, as is the next change. In fact, in all of the places where you currently have an early exit, adding this assertion seems like the right thing to do as well.